### PR TITLE
move loopback configuration to intfMgrd

### PR DESCRIPTION
**Pull request by tylerlinp**
**- What I did**
Sonic supports multiple loopback interfaces. Each loopback interfaces can belong to different Vrf.

**- How I did it**
IntfMgrd will take interface-config.service instead to handle loopback interface configuration.

**- How to verify it**
Pass vrf vs test cases and ansible test cases.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by jleveque** [comment]
Retest this please

**Comment by prsunny** [comment]
Can you please look at the build failures. You may want to fix this file `files/image_config/interfaces/interfaces.j2`

**Comment by tylerlinp** [comment]
Sorry, I made a mistake when merging.

**Comment by lguohan** [comment]
what is the pr for intfmgrd to support lo? we need both both into the buildimage at the same time.

**Comment by tylerlinp** [comment]
> eed both both into the buildimage at the sa

sonic-swss PR:
Azure/sonic-swss#943

File: files/image_config/interfaces/interfaces.j2 [hunk 1]
```diff
@@ -9,13 +9,6 @@
 # The loopback network interface
 auto lo
 iface lo inet loopback
-# Use command 'ip addr list dev lo' to check all addresses
```
Comment (nikos-github) [inline files/image_config/interfaces/interfaces.j2]: @tylerlinp @prsunny Any reason why the approach used for mgmt i/f being in the vrf can't be followed for lo? Not even sure that lo should be allowed to be in a vrf. How will redis work if lo is in a vrf?

Comment (tylerlinp) [reply files/image_config/interfaces/interfaces.j2]: > @tylerlinp @prsunny Any reason why the approach used for mgmt i/f being in the vrf can't be followed for lo? Not even sure that lo should be allowed to be in a vrf. How will redis work if lo is in a vrf?

All loopback interfaces map to device lo, that cause conflict when loopbacks belong to differents vrfs. I think lo shouldn't be add to a vrf, even though kernel don't restrict doing so. Some process which communicates locally may be affected. It is good to add a device for every loopback interface(LoopbackXXX), then lo keep original function. Then LoopbackXXX is add vrf/ip just like other l3 interface.

Comment (nikos-github) [reply files/image_config/interfaces/interfaces.j2]: Yes I understand the issue. However, you are taking away the ability to further statically configure lo with this change.

Comment (tylerlinp) [reply files/image_config/interfaces/interfaces.j2]: > Yes I understand the issue. However, you are taking away the ability to further statically configure lo with this change. The right approach to follow would be to leave lo configuration where it is and make sure other loopbacks don't map to the same loopback device.

What configuration do you mean? The ip addresses only config to kernel? If not add to ASIC, I think the addresses except 127.x.x.x & ::1 would be useless. If truely needed, maybe should filter configuration like LOOPBACK_INTERFACE|lo|1.1.1.1/32 and then add to interfaces.

**Comment by shine4chen** [comment]
@nikos-github This is the loopback device consideration which will add to upcoming vrf-hld-1.2.

- Sonic must support multiple loopback interfaces and each loopback interface can belong to different vrf. Because linux kernel can only support one lo interface. We will use dummy interface instead of lo interface in linux kernel.

The following is the example to configure new loopback interface by using dummy interface
```bash
ip link add loopback1 type dummy
ip link set loopback1 up
ip link set dev loopback1 master Vrf_blue
ip address add 10.0.2.2/32 dev loopback1
```
- In existing implementation interface-config.service takes care of loopback configuration. IntfMgrd will take interface-config.service instead to handle it.

**Comment by prsunny** [comment]
retest this please

**Comment by tylerlinp** [comment]
retest this please

**Comment by tylerlinp** [comment]
@prsunny It is better to merge this PR together with sonic-swss/943.

**Comment by prsunny** [comment]
Yes @tylerlinp. We need to update swss submodule. Can you update this PR with submodule update?

**Comment by shine4chen** [comment]
@prsunny this PR should be merged since PR 943 has been merged.

**Comment by tylerlinp** [comment]
> Yes @tylerlinp. We need to update swss submodule. Can you update this PR with submodule update?

OK, I will do it.

**Comment by prsunny** [comment]
Ok great, I saw you updated the submodule. I'll close my PR https://github.com/Azure/sonic-buildimage/pull/3720

**Submodules update here:**

85ff17d - 2019-11-07 : [VRF]: submit vrf feature (#943) [Tyler Li]
5604566 - 2019-11-06 : [vs_test] fix fdb test failed randomly (#1118) [Tyler Li]
038d994 - 2019-11-05 : [vnet]: Correct VNET route table size for BITMAP implementation (#1115) [Volodymyr Samotiy]
bb4e19c - 2019-11-04 : Not wait till kernel net_devices are created for all physical ports to (#1109) [Wenda Ni]
bab7b93 - 2019-11-02 : [portsorch] fix PortsOrch::allPortsReady() returns true when it should not (#1103) [Stepan Blyshchak]
5ab3f6b - 2019-10-31 : Updating pytest for sflow (#1095) [Sudharsan D.G]
d4ccdc3 - 2019-10-29 : Quote input strings before constructing a command line (#1098) [Qi Luo]
5516ec4 - 2019-10-29 : Check RIF/Port exists only for add entries (#1110) [Prince Sunny]
59440f2 - 2019-10-29 : Allow buffer profile apply after init (#1099) [Wenda Ni]

**Comment by tylerlinp** [comment]
retest this please

**Comment by tylerlinp** [comment]
@prsunny It seems that there is problem to build mock_tests when building normal image, not generate Makefile but want to make.

**Comment by prsunny** [comment]
retest broadcom please

**Comment by prsunny** [comment]
retest mellanox please

**Comment by prsunny** [comment]
@tylerlinp , we are working on the build failure

**Comment by prsunny** [comment]
https://github.com/Azure/sonic-swss/issues/1122

**Comment by prsunny** [comment]
@tylerlinp , can you update PR with this commit - https://github.com/Azure/sonic-swss/pull/1123 ?

**Comment by shine4chen** [comment]
retest broadcom please

**Comment by lguohan** [comment]
check #3733

**Comment by tylerlinp** [comment]
The commits have been cherry-picked to PR [#3733](https://github.com/Azure/sonic-buildimage/pull/3733) which has been merged. That's great, so this PR can be closed directlly.

**Comment by tylerlinp** [comment]
retest vs please

**Comment by prsunny** [comment]
Closing in favor of #3733

**Pull request closed without merging**